pouchdb-core/pouchdb-core.d.ts

```diff
@@ -133,10 +196,228 @@ declare namespace PouchDB {
+            include_docs?: boolean;
+
+            /** Maximum number of documents to return. */
+            limit?: number | false;
+
+            /** Include conflicts. */
+            conflicts?: boolean;
+
+            /** Include attachments. */
+            attachments?: boolean;
+
+            /** Return attachment data as Blobs/Buffers, instead of as base64-encoded strings. */
+            binary?: boolean;
+
+            /** Reverse the order of the output documents. */
+            descending?: boolean;
+
+            /** 
+             * For http adapter only, time in milliseconds for server to give a heartbeat to keep long connections open. 
+             * Defaults to 10000 (10 seconds), use false to disable the default. 
+             */
+            heartbeat?: number | false;
+
+            /**
+             * Reference a filter function from a design document to selectively get updates. 
+             * To use a view function, pass '_view' here and provide a reference to the view function in options.view. 
+             * See filtered changes for details.
+             */
+            filter?: string | {(doc: any, params: any): any};
+
+            /** Only show changes for docs with these ids (array of strings). */
+            doc_ids?: string[];
+
+            /**
+             * Object containing properties that are passed to the filter function, e.g. {"foo:"bar"}, 
+             * where "bar" will be available in the filter function as params.query.foo. 
+             * To access the params, define your filter function like function (doc, params).
+             */
+            query_params?: {[paramName: string]: any};
+
+            /**
+             * Specify a view function (e.g. 'design_doc_name/view_name' or 'view_name' as shorthand for 'view_name/view_name') to act as a filter. 
+             * Documents counted as “passed” for a view filter if a map function emits at least one record for them. 
+             * Note: options.filter must be set to '_view' for this option to work.
+             */
+            view?: string;
+        }
+
+        interface ChangesResponseChange<Content extends Core.Encodable> {
+            id: string;
+            seq: number;
+            changes: { rev: string }[];
+            deleted?: boolean;
+            doc?: ExistingDocument<Content & ChangesMeta>;
+        }
+
+        interface ChangesResponse<Content extends Core.Encodable> {
+            status: string;
+            last_seq: number;
+            results: ChangesResponseChange<Content>[];
+        }
+
+        interface Changes<Content extends Core.Encodable> extends EventEmitter {
+            on(event: 'change', listener: (value: ChangesResponseChange<Content>) => any): this;
+            on(event: 'complete', listener: (value: ChangesResponse<Content>) => any): this;
+            on(event: 'error', listener: (value: any) => any): this;
+
+            cancel(): void;
+        }
+
+        interface ReplicateOptions {
```

Comment:
This (and the rest of the replication typings) should go into a new `pouchdb-replication`, rather than `pouchdb-core`.  I can send a second PR for that if you'd prefer.